2020/CG-04-28.md

```diff
@@ -48,4 +46,238 @@ Installation is required, see the calendar invite.
+ 
+AR: One is how to opt out, the other is how to import them.. <example with trap> It’s not really a trap, we should report it as a different thing that can be defined, the other is how do we opt in/out. We specify the preconditions, and then say that there are some ways to check some preconditions but not all
+ 
+DG: If you can't check all of them, then you can't have a JS API.
+ 
+AR: The JS API may be more restricted, why would you hold everyone hostage to what the JS API does? 
+ 
+DG: Poll is: should the wasm C API, accept the PR as laid out...
+ 
+AR: Poll is too binary
+ 
+DG: Happy to work through details there, trying to get a clarity of direction..
+ 
+AR: You haven’t motivated why these changes are required? 
+ 
+Poll: Should wasm-c-api pursue safety, as discussed in this issue?
+ 
+|SF|F|N|A|SA|
+|--|-|-|-|--|
+|6|10|18|2|0|
+ 
+BS: Seems pretty clear that the group agrees that there is consensus on having safety..
+ 
+AR: The way that it’s currently done in the PR is.. I want to ensure that we have a toggle that still has the defined semantics - 
+ 
+DG: I think we fundamentally disagree here..
+ 
+BS: We can get to the point where we have to figure out what the semantics are, that sounds like a reasonable compromise.
+ 
+<More discussion DG, LW, AR - taking it offline>
+ 
+## Flexible vectors [slides](https://github.com/penzn/wasm-long-vectors), [design/1332](https://github.com/WebAssembly/design/issues/1332). Poll: Approve phase 0 for flexible vectors.
+ 
+PP: presenting <TODO slides>
+ 
+AR: One basic question: you're proposing introducing implementation-dependent behavior to wasm?
+ 
+PP: Depends on what you mean by implementation dependent? 
+ 
+RT: Doesn’t length change on each implementation? 
+ 
+PP: Yes, the vector can change on cell phone, desktop, etc.
+ 
+TL: Vector length itself is implementation defined..
+ 
+PP: Machine code you’re generating changes though
+ 
+TL: The code you're intended to generate would not have observable differences, but using this proposal you could construct an example that has different behavior on each platform, for example returning the size of the vector.
+ 
+AR: The number of
+ 
+DG: There are definitely implementation-defined aspects. Other implementations don't have to have the maximum length. The poll that Petr is putting in is for phase 0/1. Not set in stone, mostly investigative.
+ 
+KM: Is the reason why this is better... it makes code generation easier, since your loop can be dependent on what the machine supports?
+ 
+PP: Yeah, generally because you would have to.. We don’t actually combine them 
+ 
+KM You could do a loop invariant strength reduction, there are loops you can write that you don’t have to flatten it out.. 
+ 
+AR: Similar concern as KM
+ 
+JP: Consensus requirements are low: do members believe that the feature is in-scope will be workable.
```

Comment:
I think this was me, not Jay?

Reply:
Yes, this was - it's a case of conflicting initials for JP Sugarbroad and Jay Phelps. We usually try to disambiguate in the notes with when listing attendees if there are multiple folks with the same initials speaking at the meeting, but it's not always easy to keep track. Feel free to submit a PR to disambiguate (using JS, or whatever other way you would prefer).

Reply:
Yep, this most commonly happens with Deepti Gandluri and Dan Gohman too... :)